src/main/java/com/linecorp/armeria/server/docs/ThriftDocString.java

```diff
@@ -0,0 +1,172 @@
+        return cached.computeIfAbsent(classLoader, loader -> parseDocStrings(loader, getAllThriftJsons(loader)));
+    }
+
+    /**
+     * Parses DocStrings from input Thrift IDL JSON Resources.
+     * @return a map with key is FQCN and value is document string.
+     */
+    static Map<String, String> parseDocStrings(ClassLoader classLoader, Collection<String> jsonPaths) {
+        final ImmutableMap.Builder<String, String> docStrings = ImmutableMap.builder();
+        for (String jsonPath : jsonPaths) {
+            final Map<String, String> values = getDocStringsFromJsonResource(classLoader, jsonPath);
+            if (values != null) {
+                docStrings.putAll(values);
+            }
+        }
+        return docStrings.build();
+    }
+
+    static Collection<String> getAllThriftJsons(ClassLoader classLoader) {
+        final Reflections reflections = new Reflections(new ConfigurationBuilder()
+                                                                .setUrls(ClasspathHelper.forPackage(""))
+                                                                .addClassLoader(classLoader)
+                                                                .addScanners(new ResourcesScanner()));
+        return Collections2.filter(reflections.getResources(FILENAME_FILTER),
+                                   fullPath -> FULLPATH_FILTER.matcher(fullPath).matches());
+    }
+
+    /**
+     * Gets the namespace key of names.
+     * @param names name list.
+     * @return merged key.
+     */
+    static String key(String... names) {
+        return Stream.of(names).filter(s -> !Strings.isNullOrEmpty(s)).collect(Collectors.joining(DELIM));
+    }
+
+    private static Map<String, String> attrsToMap(XMLStreamReader reader) {
+        final ImmutableMap.Builder<String, String> map = ImmutableMap.builder();
+        for (int i = 0; i < reader.getAttributeCount(); i++) {
+            map.put(reader.getAttributeLocalName(i), reader.getAttributeValue(i));
+        }
+        return map.build();
+    }
+
+    private static <T> T child(Map<String, Object> map, String... keys) {
+        Object value = map;
+        for (String key : keys) {
+            value = ((Map<String, Object>) value).get(key);
+            if (value == null) {
+                return null;
+            }
+        }
+        return (T) value;
+    }
+
+    static Map<String, String> getDocStringsFromJsonResource(ClassLoader classLoader, String jsonResourcePath) {
+        ImmutableMap.Builder<String, String> docStrings = ImmutableMap.builder();
+        try (InputStream in = classLoader.getResourceAsStream(
+                Objects.requireNonNull(jsonResourcePath, "xmlResourcePath"))) {
+            if (in == null) {
+                return null;
+            }
+
+            final Map<String, Object> json = new ObjectMapper().readValue(in, JSON_VALUE_TYPE);
+            final String packageName = child(json, "namespaces", "java");
```

Comment:
I think the XML may have been more complicated, now that it's just Maps, don't need this confusing child method,

json.getOrDefault("namespaces", ImmutableMap.of()).get("java");